app/src/main/java/org/hackillinois/android/view/ScannerFragment.kt

```diff
@@ -1,215 +1,145 @@
-        override fun possibleResultPoints(resultPoints: MutableList<ResultPoint>?) {
-        }
-
-        override fun barcodeResult(result: BarcodeResult) {
-            if (shouldParseScan) {
-                // stop parsing scans until we have finished this one!
-                shouldParseScan = false
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.fragment_scanner, container, false)
 
-                val currentEvent = eventListView.selectedItem.toString()
+        view.eventListView.onItemSelectedListener = viewModel.onEventSelected
 
-                // Prevent duplicate scans
-                if (getUserIdFromQrString(result.text) == lastSuccessfullyScannedUser && currentEvent == lastEventScannedFor) {
-                    return
+        context?.let { context ->
+            codeScanner = CodeScanner(context, view.codeScannerView).apply {
+                camera = CodeScanner.CAMERA_BACK
+                formats = listOf(BarcodeFormat.QR_CODE)
+                autoFocusMode = AutoFocusMode.SAFE
+                scanMode = ScanMode.SINGLE
+                isAutoFocusEnabled = true
+                isFlashEnabled = false
+                decodeCallback = DecodeCallback {
+                    val event = view.eventListView.selectedItem.toString()
+                    val userId: String = getUserIdFromQrString(it.text)
+                    viewModel.checkUserIntoEvent(event, userId, view.staffOverrideSwitch.isChecked)
+                    Handler(Looper.getMainLooper()).post {
+                        Toast.makeText(context, "Sending request to API. Please wait for a response", Toast.LENGTH_LONG).show()
+                    }
                 }
+                errorCallback = ErrorCallback {
+                    Handler(Looper.getMainLooper()).post {
+                        Toast.makeText(context, "Camera initialization error: ${it.message}", Toast.LENGTH_LONG).show()
+                    }
+                }
+            }
 
-                val userId: String = getUserIdFromQrString(result.text)
-
-                lastEventScannedFor = eventListView.selectedItem.toString()
-
-                // Check-in is a special event in the HackIllinois API
-                if (lastEventScannedFor == CHECK_IN_TEXT) {
-                    val staffOverride = staffOverrideSwitch.isChecked
-                    val hasCheckedIn = true
-                    val hasPickedUpSwag = true
-                    val checkIn = CheckIn(userId, staffOverride, hasCheckedIn, hasPickedUpSwag)
-
-                    viewModel.checkInUser(checkIn)
+            view.codeScannerView.setOnClickListener { codeScanner.startPreview() }
 
+            // get camera permission
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                if (context.checkSelfPermission(Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
```

Comment:
Unify these with &&, then you'll only need one `else`

Reply:
👍